python/ray/tune/tuner.py

```diff
@@ -0,0 +1,87 @@
+from ray.tune.result_grid import ResultGrid
+from ray.util import PublicAPI
+
+
+# The magic key that is used when instantiating Tuner during resume.
+_TUNER_INTERNAL = "_tuner_internal"
+_SELF = "self"
+
+
+# TODO(xwjiang): Change to alpha
+@PublicAPI(stability="beta")
+class Tuner:
+    """The external facing Tuner class as part of Ray MLC effort.
+
+    Args:
+        trainable: The trainable to be tuned.
+        param_space: Search space to run tuning on.
+            One thing to note is that both preprocessor and dataset can be tuned here.
+        tune_config: Tune algo specific configs.
+            Refer to ray.tune.tune_config.TuneConfig for more info.
+        run_config: Run time configuration that is universal between
+            Tune and Train.
+
+    Returns:
+        ``ResultGrid`` object.
```

Comment:
Document what happens if an error occurs? Does the grid contain error results?

Reply:
See comment above.